argon2/src/error.rs

```diff
@@ -121,4 +121,25 @@ impl From<Error> for password_hash::Error {
 }
 
 #[cfg(feature = "std")]
-impl std::error::Error for Error {}
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Self::AdTooLong => None,
+            Self::AlgorithmInvalid => None,
+            Self::B64Encoding(err) => Some(err),
+            Self::KeyIdTooLong => None,
+            Self::MemoryTooLittle => None,
+            Self::MemoryTooMuch => None,
+            Self::OutputTooShort => None,
+            Self::OutputTooLong => None,
+            Self::PwdTooLong => None,
+            Self::SaltTooShort => None,
+            Self::SaltTooLong => None,
+            Self::SecretTooLong => None,
+            Self::ThreadsTooFew => None,
+            Self::ThreadsTooMany => None,
+            Self::TimeTooSmall => None,
+            Self::VersionInvalid => None,
+        }
```

Comment:
Really this can just be...

```suggestion
        match self {
            Self::B64Encoding(err) => Some(err),
            _ => None
        }
```

...and now that I mention it, this is also true of https://github.com/RustCrypto/traits/pull/1264/files#diff-c39a2e2913ce6aa08f2b5ae5c168f57989e954720d0bc3e587a893ce07dc2e3a

Reply:
Indeed, I can simplify it.
A potential downside is that added variants with wrapped errors will by default return `None`, whereas with this manual listing new variants will trigger an error until added on purpose.

Reply:
There is that, I suppose